Src/LinearSolvers/Projections/AMReX_NodalProjector.cpp

```diff
@@ -296,31 +290,20 @@ NodalProjector::project ( Real a_rtol, Real a_atol )
         //
         // vel = vel + fluxes = vel - ( sigma / alpha ) * grad(phi)
         //
-        // Since we already averaged-down the velocity field and -grad(phi),
-        // we perform the projection by simply adding the two of them.
-        // In virtue of the linearity of the operations involved, this is equivalent
-        // to averaging down the velocity only once AFTER summing -grad(phi)
-        //
         MultiFab::Add( *m_vel[lev], m_fluxes[lev], 0, 0, AMREX_SPACEDIM, 0);
 
         // set m_fluxes = grad(phi)
-        m_fluxes[lev].mult(-1.0);
-        for (int n = 0; n < AMREX_SPACEDIM; ++n)
-        {
-            if (m_has_alpha)
-            {
-                MultiFab::Multiply(m_fluxes[lev], *m_alpha[lev], 0, n, 1, 0);
-            }
-            if (m_sigma.empty()) {
-                AMREX_ASSERT(m_const_sigma != Real(0.));
-                m_fluxes[lev].mult(Real(1.)/m_const_sigma, n, 1, 0);
-            } else {
-                MultiFab::Divide(m_fluxes[lev], *m_sigma[lev], 0, n, 1, 0);
-            }
-        }
-
+	m_linop->compGrad(lev,m_fluxes[lev],m_phi[lev]);
     }
 
+    //
+    // At this time, results are "correct" only on regions not covered by finer grids.
+    // We average them down so that they are "correct" everywhere in each level.
+    //
+    averageDown(GetVecOfPtrs(m_fluxes));
```

Comment:
Is this necessary for non-EB?  When `LinOp::compGrad` is called, the nodal phi has been averaged down.  So I don't think this is necessary at least for non-EB.

Reply:
Looking at the numbers says we do really need to average down here. ... For nodal, doesn't average down just copy the overlying fine node? Whereas gradPhi is cell centered, so it does a more traditional averaging, which would result in non-overlying nodes getting included.

Reply:
Then something is not quite right.

Reply:
Yeah. I was just thinking the same thing

Reply:
Actually you are right.  They are supposed to be different.